cloudflare_workers/loadbalancer.ts

```diff
@@ -17,45 +17,58 @@ async function fetchWithTimeout(resource: RequestInfo, options: RequestOptions =
   return response
 }
 
-async function handleRequest(request: Request): Promise<Response> {
-  const primaryUrl = new URL(request.url)
-  primaryUrl.hostname = MAIN_HOST
-  let res: Response
-  try {
-    res = await fetchWithTimeout(primaryUrl.toString(), request.clone())
-  }
-  catch (err) {
-    console.log('err', primaryUrl, err)
-    // try the second host
-    const backupUrl = new URL(request.url)
-    backupUrl.hostname = BACKUP_HOST
-    const end = backupUrl.pathname.split('/').pop()
-    // https://web.capgo.app/api-egde/ok
-    backupUrl.pathname = `/api-egde/${end}`
+export default {
+  async fetch(request: Request) {
+    let res: Response
+    const primaryUrl = new URL(request.url)
+    primaryUrl.hostname = MAIN_HOST
     try {
-      // Use fetchWithTimeout for the backup requests as well
-      res = await fetchWithTimeout(backupUrl.toString(), request.clone())
+      res = await fetchWithTimeout(primaryUrl.toString(), {
+        redirect: 'follow',
+        body: request.body,
+        method: request.method,
+        headers: request.headers,
```

Comment:
why do not keep the `request.clone()` and add the only redirect?

```suggestion
        ...request.clone(),
        redirect: 'follow',
```

we can do the same on lines 43 and 59, also

Reply:
another possible option would be to create the object first, instead of creating it on each `fetchWithTimeout` (this is preferably than my first suggestion)